plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/DeleteFile.java

```diff
@@ -40,18 +35,20 @@ public record DeleteFile(
         long recordCount,
         long fileSizeInBytes,
         List<Integer> equalityFieldIds,
-        Map<Integer, byte[]> lowerBounds,
-        Map<Integer, byte[]> upperBounds,
+        Optional<Long> rowPositionLowerBound,
+        Optional<Long> rowPositionUpperBound,
         long dataSequenceNumber)
 {
     private static final long INSTANCE_SIZE = instanceSize(DeleteFile.class);
 
     public static DeleteFile fromIceberg(org.apache.iceberg.DeleteFile deleteFile)
     {
-        Map<Integer, byte[]> lowerBounds = firstNonNull(deleteFile.lowerBounds(), ImmutableMap.<Integer, ByteBuffer>of())
-                .entrySet().stream().collect(toImmutableMap(Map.Entry::getKey, entry -> getWrappedBytes(entry.getValue()).clone()));
-        Map<Integer, byte[]> upperBounds = firstNonNull(deleteFile.upperBounds(), ImmutableMap.<Integer, ByteBuffer>of())
-                .entrySet().stream().collect(toImmutableMap(Map.Entry::getKey, entry -> getWrappedBytes(entry.getValue()).clone()));
+        Optional<Long> rowPositionLowerBound = Optional.ofNullable(deleteFile.lowerBounds())
+                .map(bounds -> bounds.get(DELETE_FILE_POS.fieldId()))
```

Comment:
do we need to filter out nulls here ?
i'm not sure that having bounds for DELETE_FILE_POS field is always guaranteed

Reply:
Yeah, better safe than sorry. Don't want to end up with an Optional.of(null) value

Reply:
Actually I was wrong about that, if bounds doesn't have an entry for the DELETE_FILE_POS column it'll return null. Optional.map converts null returns to `empty`